In [4]:
import os

train_dir = r'C:\TFG\Dataset_Final_Procesado\Dataset_Final_Procesado\train'

archivos = []
for root, dirs, files in os.walk(train_dir):
    for f in files:
        archivos.append(os.path.join(root, f))  # ruta completa
        if len(archivos) >= 20:
            break
    if len(archivos) >= 20:
        break

if not archivos:
    print("❌ No se encontraron archivos. Verifica que la ruta existe.")
    print("¿Existe la carpeta?", os.path.exists(train_dir))
    
    # Muestra qué hay dentro
    print("\nContenido de la ruta:")
    for item in os.listdir(train_dir):
        print(" →", item)
else:
    print(f"✅ Se encontraron archivos. Mostrando primeros {len(archivos)}:\n")
    for a in archivos:
        print(a)

✅ Se encontraron archivos. Mostrando primeros 20:

C:\TFG\Dataset_Final_Procesado\Dataset_Final_Procesado\train\fake\DeepFakeDetection_22_23__outside_talking_pan_laughing__TU7RHG0S_f0.jpg
C:\TFG\Dataset_Final_Procesado\Dataset_Final_Procesado\train\fake\DeepFakeDetection_22_23__outside_talking_pan_laughing__TU7RHG0S_f1.jpg
C:\TFG\Dataset_Final_Procesado\Dataset_Final_Procesado\train\fake\DeepFakeDetection_22_23__outside_talking_pan_laughing__TU7RHG0S_f10.jpg
C:\TFG\Dataset_Final_Procesado\Dataset_Final_Procesado\train\fake\DeepFakeDetection_22_23__outside_talking_pan_laughing__TU7RHG0S_f11.jpg
C:\TFG\Dataset_Final_Procesado\Dataset_Final_Procesado\train\fake\DeepFakeDetection_22_23__outside_talking_pan_laughing__TU7RHG0S_f12.jpg
C:\TFG\Dataset_Final_Procesado\Dataset_Final_Procesado\train\fake\DeepFakeDetection_22_23__outside_talking_pan_laughing__TU7RHG0S_f13.jpg
C:\TFG\Dataset_Final_Procesado\Dataset_Final_Procesado\train\fake\DeepFakeDetection_22_23__outside_talking_pan_laughing__TU

# Comprobaciones

In [5]:
import os

train_dir = r'C:\TFG\Dataset_Final_Procesado\Dataset_Final_Procesado\train'
test_dir  = r'C:\TFG\Dataset_Final_Procesado\Dataset_Final_Procesado\test'

def extraer_id_video(nombre_archivo):
    # Elimina la extensión y separa por "_f" para quedarse con el ID del vídeo
    sin_ext = os.path.splitext(nombre_archivo)[0]  # quita .jpg
    # El ID es todo menos la última parte "_fX"
    partes = sin_ext.rsplit('_f', 1)  # divide por el ÚLTIMO "_f"
    return partes[0]  # ej: "DeepFakeDetection_22_23__outside_talking_pan_laughing__TU7RHG0S"

# --- Recoger IDs de train ---
ids_train = set()
for root, dirs, files in os.walk(train_dir):
    for f in files:
        if f.lower().endswith(('.jpg', '.png')):
            ids_train.add(extraer_id_video(f))

# --- Recoger IDs de test ---
ids_test = set()
for root, dirs, files in os.walk(test_dir):
    for f in files:
        if f.lower().endswith(('.jpg', '.png')):
            ids_test.add(extraer_id_video(f))

# --- Comprobar solapamiento ---
solapamiento = ids_train & ids_test

print(f"Vídeos únicos en train: {len(ids_train)}")
print(f"Vídeos únicos en test:  {len(ids_test)}")
print(f"Vídeos en AMBOS splits: {len(solapamiento)}")

if len(solapamiento) == 0:
    print("\n✅ Split correcto: ningún vídeo está en train y test a la vez.")
    print("   Tu entrenamiento NO tiene data leakage.")
else:
    print(f"\n❌ HAY DATA LEAKAGE en {len(solapamiento)} vídeos.")
    print("Ejemplos de vídeos repetidos:")
    for v in list(solapamiento)[:5]:
        print(f"  → {v}")

Vídeos únicos en train: 1596
Vídeos únicos en test:  398
Vídeos en AMBOS splits: 0

✅ Split correcto: ningún vídeo está en train y test a la vez.
   Tu entrenamiento NO tiene data leakage.


1596 × 15 frames = 23.940 frames (entrenamiento)

398 vídeos en test → 398 × 15 frames = 5.970 frames (test)

Ratio exacto: 1596 / (1596+398) = 80.08% → es exactamente 80/20